src/nuclio/common/services/export.service.js

```diff
@@ -0,0 +1,188 @@
+            };
+
+            var blob = prepareBlobObject(functionToExport);
+
+            downloadExportedFunction(blob, version.metadata.name);
+        }
+
+        /**
+         * Exports the project
+         * @param {Object} project
+         * @param {Function} getFunctions
+         */
+        function exportProject(project, getFunctions) {
+            getFunctions({id: project.metadata.name})
+                .then(function (functions) {
+                    var functionsList = lodash.map(functions, function (functionItem) {
+                        return lodash.chain(functionItem)
+                            .set('spec.version', 1)
+                            .omit(['status', 'metadata.namespace'])
+                            .value();
+                    });
+
+                    var projectToExport = {
+                        project: {
+                            metadata: {
+                                name: project.spec.displayName
+                            },
+                            spec: {
+                                functions: functionsList
+                            }
+                        }
+                    };
+
+                    var blob = prepareBlobObject(projectToExport);
+
+                    downloadExportedFunction(blob, project.spec.displayName);
+                })
+                .catch(function (error) {
+                    var msg = 'Oops: Unknown error occurred while exporting the project';
+                    DialogsService.alert(lodash.get(error, 'data.error', msg));
+                });
+
+        }
+
+        /**
+         * Exports projects
+         * @param {Object} projects
+         * @param {Function} getFunctions
+         */
+        function exportProjects(projects, getFunctions) {
+            var promises = lodash.map(projects, function (project) {
+                return getFunctions({id: project.metadata.name})
+                    .then(function (functions) {
+                        var functionsList = lodash.map(functions, function (functionItem) {
+                            return lodash.chain(functionItem)
+                                .set('spec.version', 1)
+                                .omit(['status', 'metadata.namespace'])
+                                .value();
+                        });
+
+                        return {
+                            metadata: {
+                                name: project.spec.displayName
+                            },
+                            spec: {
+                                functions: functionsList
+                            }
+                        };
+                    })
+                    .catch(angular.noop);
+            });
+
+            $q.all(promises)
+                .then(function (projectsToExport) {
+                    var blob = prepareBlobObject({
+                        projects: projectsToExport
```

Comment:
If a promise from above fails, it will reach `.catch(angular.noop)` which will result in resolving the promise with the value `undefined`.
This means that in this line here, `projectsToExport` might look like this:
```js
[
    {
        metadata: {
            name: 'project-name1'
        },
        spec: {
            functions: [ ... ]
        }
    },
    undefined, // <-- failed promise
    {
        metadata: {
            name: 'project-name3'
        },
        spec: {
            functions: [ ... ]
        }
    },
    {
        metadata: {
            name: 'project-name4'
        },
        spec: {
            functions: [ ... ]
        }
    },
    undefined  // <-- failed promise
]
```
This is why I suggest using `lodash.compact` in order to filter out these `undefined` values:
```suggestion
                        projects: lodash.compact(projectsToExport)
```